clients/src/main/java/org/apache/kafka/clients/consumer/internals/CommitRequestManager.java

```diff
@@ -149,6 +185,29 @@ Queue<StagedCommit> stagedCommits() {
         return this.stagedCommits;
     }
 
+    /**
+     * Get all the sendable requests, and create a list of UnsentRequest.
+     */
+    private List<NetworkClientDelegate.UnsentRequest> sendOffsetFetchRequests(final long currentTimeMs) {
+        List<UnsentOffsetFetchRequestState> requests = unsentOffsetFetchRequests.sendableRequests(currentTimeMs);
+        List<NetworkClientDelegate.UnsentRequest> pollResults = new ArrayList<>();
+        requests.forEach(req -> {
+            OffsetFetchRequest.Builder builder = new OffsetFetchRequest.Builder(
+                    groupState.groupId, true,
+                    new ArrayList<>(req.requestedPartitions),
+                    throwOnFetchStableOffsetUnsupported);
+            NetworkClientDelegate.UnsentRequest unsentRequest = new NetworkClientDelegate.UnsentRequest(
+                    builder,
+                    coordinatorRequestManager.coordinator());
+            FetchCommittedOffsetResponseHandler cb = new FetchCommittedOffsetResponseHandler(req);
+            unsentRequest.future().whenComplete((r, t) -> {
```

Comment:
When we first made the FutureCompletionHandler, we explicitly handled the responses in the onComplete method here:
```            if (response.authenticationException() != null) {
                onFailure(response.authenticationException());
            } else if (response.wasDisconnected()) {
                onFailure(DisconnectException.INSTANCE);
            } else if (response.versionMismatch() != null) {
                onFailure(response.versionMismatch());
            } else {
                future.complete(response);
            }
```
            
I was thinking about your suggestion, but since you mentioned it again, let me try to refactor this a bit to make it more pluggin-able

Reply:
I think we need this logic to handle the disconnect.  This is actually from here: https://github.com/apache/kafka/blob/3d2ac7cdbe89cdabfd95db9971de31878afa5498/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java#L605